### PR TITLE
add prefix (prepend to url) - test, docs (new)

### PR DESCRIPTION
**Pull request by whitecolor**
New version does not change url in reqOptions, fixes https://github.com/cyclejs/cycle-http-driver/issues/35

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -19,6 +19,7 @@ follow a structure similar to superagent's request API itself.
 `request` object properties:
 
 - `url` *(String)*: the remote resource path. **required**
+- `prefix` *(String)*: prepends url with value, overrides driver option.
```
Comment (staltz) [inline docs/api.md]: Can I ask you to move these changes out of this file and into src/index.js? This file is auto-generated from that, so we never should directly modify this one.

File: src/http-driver.js [hunk 2]
```diff
@@ -67,9 +72,9 @@ function urlToSuperagent(url) {
   return superagent.get(url)
 }
 
-function createResponse$(reqOptions) {
+function createResponse$(reqOptions, options) {
```
Comment (staltz) [inline src/http-driver.js]: Why this second param?

Comment (whitecolor) [inline src/http-driver.js]: Oh, It was from previous version, forgot to remove.

File: docs/api.md (hunk 1 above)
Comment (whitecolor) [inline docs/api.md]: ok didn't notice.

**Comment by whitecolor** [comment]
Just thought. Maybe if this can be accomplished easily by user, there is no need to add this feature to driver.

**Comment by staltz** [comment]
@whitecolor what do you think? Which alternative are you more likely to choose? Not have prefix feature or to have it?

**Comment by whitecolor** [comment]
I think I would prefere to have it available out of the box just to allow more simple usage. If user will need more sophisticated logic for mutating urls he/she will do it inside the app, not the driver.

**Comment by whitecolor** [comment]
So this actually allow to get rid of folloing
```javascript
request$ = some$.map(() => `${hostURL}/api/some`)
```

if routes are spread acrosss the modules one would need access to `prefixURL` in every module. Or it can be done with boilerplate wrapper, which will add all prefixes, etc:
```javascript
request$ = some$.map(() => makeRequest(`/some`))
```
The same problem with filtering incoming responses by url.

Or you can just put in one place:
```javascript
  makeDOMDriver({prefix: '${hostURL}/api'})
```
and urls provided in requests will be kept short and managed more simply.

**Comment by staltz** [comment]
@whitecolor I gave some serious thinking about this feature, and I think we need to maintain one guarantee in Cycle.js and across all standard drivers: "app/business logic only inside `main`". The options that are given to `makeDriver` functions are supposed to configure or specify what *variant* of effects we want as a target. E.g. `makeHTTPDriver({eager: true})` is not app logic, it's specifying a different "kind" of HTTP Driver, handling effects in a slightly different manner.

I can see how we can easily start creeping some bits of app logic into drivers, so we need a conscious effort to avoid it. Ideally, one only needs to look at `main` in order to understand an application. Driver configurations are just there because they need to exist in order to accept the sinks that the `main` produces.

Thanks a lot for the effort, even though I'll close this PR you can be sure I appreciate them and will welcome more PRs in the future too.

**Comment by whitecolor** [comment]
I'm ok with that desision, I too was doubt that it is correct thing to add.

**Comment by staltz** [comment]
PS: I got convinced that if someone wants a prefix, they should add pre/post processing steps in main, like you pointed out:

```js
const finalRequest$ = request$.map(suffix => `${prefixURL}/suffix`);
```
Just before returning sinks.

And
```js
const response$$ = sources.HTTP.map(response$ => {
  response$.request = {
    ...response$.request,
    url: response$.request.url.replace(prefixURL, ''),
  }
  return response$;
});
```
At the top of `main`, when getting the sources.

**Comment by whitecolor** [comment]
Yeah I think this is a correct approach)

**Pull request closed without merging**